pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,183 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+	isTrue                 = "True"
+)
+
+// ClusterOperatorBuilder provides struct for ClusterOperator object.
+type ClusterOperatorBuilder struct {
+	// ClusterOperator definition. Used to create a clusteroperator object.
+	Definition *v1.ClusterOperator
+	// Created clusteroperator object.
+	Object *v1.ClusterOperator
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate ClusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Exists checks whether the given ClusterOperator exists.
+func (builder *ClusterOperatorBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if clusteroperator %s exists", builder.Definition.Name)
+
+	err := builder.Discover()
```

Comment:
Please just use the versioned client get method directly for this 
```
builder.apiClient.ClusterOperators().Get()
```
Review pod pkgs for [example](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/pod/pod.go#L494)